adcc/OneParticleOperator.py

```diff
@@ -52,41 +57,183 @@ def __init__(self, spaces, is_symmetric=True):
             assert isinstance(spaces.reference_state, libadcc.ReferenceState)
             self.reference_state = spaces.reference_state
 
-    @classmethod
-    def from_cpp(cls, cpp_operator):
-        assert cpp_operator.cartesian_transform == "1"
-        ret = cls(cpp_operator.mospaces, cpp_operator.is_symmetric)
-        for b in cpp_operator.blocks_nonzero:
-            ret.set_block(b, cpp_operator.block(b))
+        occs = sorted(self.mospaces.subspaces_occupied, reverse=True)
+        virts = sorted(self.mospaces.subspaces_virtual, reverse=True)
+        self.orbital_subspaces = occs + virts
+        if self.is_symmetric:
+            combs = list(
+                combinations_with_replacement(self.orbital_subspaces, r=2)
+            )
+        else:
+            combs = list(product(self.orbital_subspaces, repeat=2))
```

Comment:
Can you explain in a comment why you are doing it exactly like this?

Reply:
Sure!